tests/DCPS/LargeSample/DataReaderListener.cpp

```diff
@@ -215,8 +215,12 @@ bool DataReaderListenerImpl::data_consistent() const
                   << writer->second.size() << std::endl;
         valid_and_done = false;
         if (writer->second.size() < samples_per_writer_) {
-          for (size_t sample = 0; sample < samples_per_writer_; ++sample) {
-            std::cerr
+#if __cplusplus >= 199707L
```

Comment:
Why is this conditional?  Isn't `samples_per_writer_` always a `size_t`?

Reply:
Oh, I think you're right.  I'm at a disadvantage when trying to fix warnings in that my Visual Studio install stupidly doesn't reproduce them.  It's the passing of 'sample' to 'writer->second.count(sample)', and to 'expected_data_field_length' that is producing the warnings.  'count' wants a const int.  'expected_data_field_length' wants an int.

Reply:
I modified this to cast the size_t to the types being complained about.  I was able to get these warnings to show up in my visual studio by changing the DDS_LargeSample_Subscriber project settings to show all warnings.  I'm now seeing these warnings, which this change fixed, along with lots of other warnings.